diff --git a/pkgs/development/compilers/llvm/common/compiler-rt/default.nix b/pkgs/development/compilers/llvm/common/compiler-rt/default.nix
--- a/pkgs/development/compilers/llvm/common/compiler-rt/default.nix
+++ b/pkgs/development/compilers/llvm/common/compiler-rt/default.nix
@@ -31,12 +31,17 @@
   # making them part of builtins.a. Unless no dynamic linking is used at
   # all, this is the correct setup as it ensures the locks are unique in
   # memory.
-  withAtomicsLib ? stdenv.hostPlatform.hasSharedLibraries,
+  withAtomicsLib ?
+    stdenv.hostPlatform.hasSharedLibraries
+    && !stdenv.hostPlatform.isWindows,
   # If withAtomics is enabled, this selects the pthreads-based
   # implementation of the routines instead of the implementation
   # using ad-hoc mutexes (which doesn't depend on libc at all).
   # Use of pthreads helps code play better with sanitizers.
-  withAtomicsPthread ? lib.versionAtLeast release_version "19" && stdenv.cc.libc != null,
+  withAtomicsPthread ?
+    lib.versionAtLeast release_version "19"
+    && stdenv.cc.libc != null
+    && !stdenv.hostPlatform.isWindows,
-
+
   # In recent releases, the compiler-rt build seems to produce
   # many `libclang_rt*` libraries, but not a single unified
